spacelift/internal/structs/module.go

```diff
@@ -36,33 +36,53 @@ type Module struct {
 // ExportVCSSettings exports VCS settings into Terraform schema.
 func (m *Module) ExportVCSSettings(d *schema.ResourceData) error {
 	var fieldName string
-	vcsSettings := make(map[string]interface{})
+	var vcsSettings map[string]interface{}
 
 	switch m.Provider {
 	case VCSProviderAzureDevOps:
-		vcsSettings["id"] = m.VCSIntegration.ID
-		vcsSettings["project"] = m.Namespace
-		vcsSettings["is_default"] = m.VCSIntegration.IsDefault
+		if m.VCSIntegration != nil {
```

Comment:
what happens if VCSIntegration is nil? Can this happen now for default integrations (or more specifically for legacy setups), or will this always be populated for stacks that are not in detached state?

Reply:
always populated for attached. always nil for detached.